easybuild/framework/easyconfig/tweak.py

```diff
@@ -1060,11 +1060,36 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
     tweaked_spec = os.path.join(targetdir or tempfile.gettempdir(), ec_filename)
 
     parsed_ec.dump(tweaked_spec, always_overwrite=False, backup=True)
+    regex_attempt_on_ecspec = attempt_regex_substitution(ec_spec, tweaked_spec)
+    if regex_attempt_on_ecspec:
+        # Prefer a simple regex substitution version
+        pass
     _log.debug("Dumped easyconfig tweaked via --try-* to %s", tweaked_spec)
 
     return tweaked_spec
 
 
+def attempt_regex_substitution(original_spec, tweaked_spec):
+    regexed_spec = None
+
+    original_ecdict = EasyConfig(original_spec).asdict()
```

Comment:
Unfortunately the templating is not always resolved in parameters such as `postinstallcmds`  in this approach so the matching is not good enough. We need to completely remove possible templates from the raw txt to be able to compare like with like. With this I get
```
alanc@~/EasyBuild_Git/easybuild-framework(regex_tweaking)$ time eb --experimental --dry-run --robot --try-toolchain=foss,2019a --try-update-deps mpiP-3.4.1-iimpi-2019a.eb
== temporary log file in case of crash /tmp/eb-73awbqw5/easybuild-n5kshsrz.log

WARNING: Your request will result in a compiler family switch (Intel to GCC). Here be dragons!


WARNING: Your request will result in a compiler family switch (Intel to GCC). Here be dragons!

{}
{}
{'toolchain': ({'name': 'iimpi', 'version': '2019a'}, {'name': 'gompi', 'version': '2019a'}), 'postinstallcmds': (["sed -e 's,MPIP_LIB_DIR=.*,MPIP_LIB_DIR=%(installdir)s/lib,' -i bin/mpirun-mpip", "sed -e 's,MPIP_DIR=.*,MPIP_DIR=%(installdir)s,' -i bin/srun-mpip"], ["sed -e 's,MPIP_LIB_DIR=.*,MPIP_LIB_DIR=%(installdir)s/lib,' -i bin/mpirun-%(namelower)s", "sed -e 's,MPIP_DIR=.*,MPIP_DIR=%(installdir)s,' -i bin/srun-%(namelower)s"])}
Dry run: printing build status of easyconfigs and dependencies

```

Reply:
It's probably some evil escaping because even if I try to resolve it by hand with `resolve_template()` it still doesn't work